### PR TITLE
[session][test] Extend TestSessionManagerDispatch with group test

### PR DESCRIPTION
**Pull request by turon**
#### Issue Being Resolved
* Fixes comment in #22736 
_Originally posted by @tcarmelveilleux in https://github.com/project-chip/connectedhomeip/pull/22736#discussion_r975342909_

#### Change overview
Adds support for groupcast test vectors for message decrypt.
Adds non-zero length payload test vector for pase unicast message decrypt.

### CODE REVIEW

File: src/transport/tests/TestSessionManagerDispatch.cpp [hunk 1]
```diff
@@ -100,19 +108,80 @@ struct MessageTestEntry theMessageTestVector[] = {
         .encryptedLength = 30,
         .privacyLength   = 30,
 
+        // NOTE: unicast message tests must use test key currently
+        .encryptKey = "\x5e\xde\xd2\x44\xe5\x53\x2b\x3c\xdc\x23\x40\x9d\xba\xd0\x52\xd2",
```
Comment (tcarmelveilleux) [inline src/transport/tests/TestSessionManagerDispatch.cpp:112]: Why?

File: src/transport/tests/TestSessionManagerDispatch.cpp [hunk 2]
```diff
@@ -100,19 +108,80 @@ struct MessageTestEntry theMessageTestVector[] = {
...
+        .name     = "secure group message (no privacy)",
+        .peerAddr = "::1",
+
+        .payload = "",
+
+        .plain     = "\06\x7d\xdb\x01\x78\x56\x34\x12\x01\x00\x00\x00\x00\x00\x00\x00\x02\x00\x01\x64\xee\x0e\x20\x7d",
+        .encrypted = "\x06\x7d\xdb\x01\x78\x56\x34\x12\x01\x00\x00\x00\x00\x00\x00\x00\x02\x00\x65\xc7\x67\xbc\x6c\xda"
+                     "\x01\x06\xc9\x80\x13\x23\x90\x0e\x9b\x3c\xe6\xd4\xbb\x03\x27\xd6", // Includes MIC
+        .privacy = "\x06\x7d\xdb\x01\x78\x56\x34\x12\x01\x00\x00\x00\x00\x00\x00\x00\x02\x00\x65\xc7\x67\xbc\x6c\xda"
+                   "\x01\x06\xc9\x80\x13\x23\x90\x0e\x9b\x3c\xe6\xd4\xbb\x03\x27\xd6", // Includes MIC
```
Comment (tcarmelveilleux) [inline src/transport/tests/TestSessionManagerDispatch.cpp:153]: Suggest you include cases where the MIC is wrong or MIC is correct but privacy key wrong, etc. There are currently no negative cases

File: src/transport/tests/TestSessionManagerDispatch.cpp (hunk 1 above)
Comment (turon) [reply src/transport/tests/TestSessionManagerDispatch.cpp:112]: It's a limitation in the current implementation of `InjectPaseSessionWithTestKey`.  I can generate an issue to track and turn the NOTE: into a TODO(#issue):

File: src/transport/tests/TestSessionManagerDispatch.cpp (hunk 2 above)
Comment (turon) [reply src/transport/tests/TestSessionManagerDispatch.cpp:153]: Yes, this PR adds group messages.  Negative is a follow-on.  I can make an issue to track.

**Review by turon — COMMENTED** [review]
I'll update to capture the suggested TODO items.

File: src/transport/tests/TestSessionManagerDispatch.cpp (hunk 2 above)
Comment (turon) [reply src/transport/tests/TestSessionManagerDispatch.cpp:153]: #22831

File: src/transport/tests/TestSessionManagerDispatch.cpp (hunk 1 above)
Comment (turon) [reply src/transport/tests/TestSessionManagerDispatch.cpp:112]: #22830

**Review by tcarmelveilleux — APPROVED** [review]

**Review by saurabhst — APPROVED** [review]

**Pull request merged**